main.js

```diff
@@ -59,7 +76,57 @@ function getCurrentGutter() {
 	return getGridProperties().gutter;
 }
 
+/**
+ * This sets MediaQueryListeners on all the o-grid breakpoints
+ * and fires a `o-grid.layoutChange` event on layout change.
+ */
+function addBreakpointListeners() {
+	try {
+		// Create a map containing all breakpoints exposed via html:before
+		const breakpoints = new Map(Object.entries(getGridBreakpoints().layouts));
+		const decr1 = val => `${Number(val.replace('px', '') - 1)}px`;
+
+		// Generate media queries for each
+		breakpoints.forEach((width, size) => {
+			const queries = Array();
+			if (size === 'S') {
+				queries.push(`(max-width: ${ width })`);
+				queries.push(`(min-width: ${ width }) and (max-width: ${ decr1(breakpoints.get('M')) })`);
+			} else if (size === 'M') {
+				queries.push(`(min-width: ${ width }) and (max-width: ${ decr1(breakpoints.get('L')) })`);
+			} else if (size === 'L') {
+				queries.push(`(min-width: ${ width }) and (max-width: ${ decr1(breakpoints.get('XL')) })`);
+			} else if (size === 'XL') {
+				queries.push(`(min-width: ${ width })`);
+			}
+
+			// matchMedia listener handler: Dispatch `o-grid.layoutChange` event if a match
+			const handleMQChange = mql => {
+				if (mql.matches) {
+					window.dispatchEvent(new CustomEvent('o-grid.layoutChange', {
+						detail: {
+							layout: size,
+						}
+					}));
+				}
+			}
+
+			// Create a new listener for each layout
+			queries.forEach(mq => {
+				const mql = window.matchMedia(mq);
+				mql.addListener(handleMQChange);
+				handleMQChange(mql);
+			});
+		});
+	} catch (e) {
+		console.error(e);
```

Comment:
@JakeChampion I used a try/catch for cases wherein `oGridSurfaceLayoutSizes` hasn't been included in Sass and thus `getGridBreakpoints()` won't return an object with a `layouts` property. If preferable, I could just add a check that bails if the response from `getGridBreakpoints()` doesn't have a `layouts` property?